integreat_cms/cms/models/abstract_tree_node.py

```diff
@@ -282,7 +287,21 @@ def move(self, target, pos=None):
         # Moving a node can modify all other nodes via raw sql queries (which are not recognized by cachalot),
         # so we have to invalidate the whole model manually.
         invalidate_model(self.__class__)
-        super().move(target=target, pos=pos)
+        try:
+            with db_mutex(f"tree-page-{self.region}"):
```

Comment:
```suggestion
            with db_mutex("page-tree"):
```
This is not sufficient, since all page trees of all regions are stored in the same database table and moving a page in one region can also change the `tree_id`s of pages in other regions (which is probably the main source of the race conditions, in most cases there aren't multiple pages moved in the same region at the same time - only because of the interference between regions, this is such a big problem).

Apart from that: Do you think the Python inbuilt [`multiprocessing.Lock`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Lock) could also be used for this task?

Reply:
Global lock: Yeah, you're right. I missed that.

About [multiprocessing.Lock](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Lock): I don't think so, as Apache2 starts the processes independently of each other (AFAICT). In the example of the documentation the lock needs to be passed to each process: https://docs.python.org/3/library/multiprocessing.html#synchronization-between-processes

Reply:
Ah, yeah sounds plausible.